### ISSUE TITLE
CocoaPods didn't install in the Build Service for Xcode export

### ISSUE THREAD

**Issue by Chadori** [Scirra/Construct-bugs#6934]
## Problem description

Hi Ashley, it seems that Cordova plugins with Pods dependencies fail to install in the Construct 3 Build Service. Have you tried updating CocoaPods in your Build Service?

## Attach a .c3p

[Test package.zip](https://github.com/Scirra/Construct-bugs/files/11243013/Test.package.zip)

**Legend:**
- `Test plugin.c3p` - Empty Construct 3 project with only the test plugin.
- `test_plugin.c3addon` - Empty Construct 3 plugin with only a reference to `onesignal-cordova-plugin`.
- `Test project.ios.project (Construct 3 export)` - Exact Xcode export from the Construct 3 Build Service without Pods being installed.
- `Test project.ios.project (Fixed Pods)` - Xcode export after applying `pod install` in the Terminal.

For your reference, these are the only lines added in the `test_plugin.c3addon`:
![image](https://user-images.githubusercontent.com/31282960/232331843-92cb1fcb-71f1-44e5-a347-1a60d84066c4.png)

## Steps to reproduce

1. Install the `test_plugin.c3addon`
2. Open the project file `Test plugin.c3p` on Construct 3.
3. Export to iOS (Xcode option)
4. Launch Xcode and open the workspace, then click the **play button**. The error below will be observed:
![image](https://user-images.githubusercontent.com/31282960/232332103-5b727e01-0b40-46a7-85be-e907b94437b8.png)

## Steps to fix

1. Open the terminal
2. `cd` to directory of the Xcode workspace folder.
3. Type in `pod install`
4. Relaunch Xcode and reopen the workspace, then click the **play button**. The build will now be successful:
![image](https://user-images.githubusercontent.com/31282960/232332215-2c5b6cd7-3c39-4136-a3b0-6fb02779461f.png)

## Observed result

CocoaPods does not install from the Build Service's cloud, for Cordova plugins with Pods dependencies. For example, here are one of the most used public Cordova plugins that are affected:
- [onesignal-cordova-plugin](https://www.npmjs.com/package/onesignal-cordova-plugin)
- [cordova-plugin-firebase-analytics](https://www.npmjs.com/package/cordova-plugin-firebase-analytics)
- [cordova-plugin-firebase-config](https://www.npmjs.com/package/cordova-plugin-firebase-config)

## Expected result

CocoaPods to install from the Construct 3 Build Service cloud.

## More details

**Affected browsers/platforms:** iOS only (Xcode export)
## System details

Construct 3 r336 - Stable

**Comment by AshleyScirra** [comment]
As noted in #6901 it works for Construct's own Mobile Advert plugin, which uses cocoapods and correctly exports. Are you sure it's not a configuration problem with the Cordova plugin? The build server is up-to-date with the latest cocoapods version (v1.12.0).

**Comment by AshleyScirra** [comment]
I spent a while looking in to this and I think it turns out that Cordova sometimes does run `pod install` automatically, and sometimes doesn't, seemingly depending on the particular configuration in use. I would guess this is some kind of Cordova bug. However it can fairly straightforwardly be worked around by getting the build server to run `pod install` if it doesn't look like Cordova has already done it. I've updated the build service to do this and the change is now live, and it looks like it's now returning Xcode builds with a `Pods` folder. Can you give it a try and see if it's working properly for you now?

**Comment by therealPaulPlay** [comment]
Great news! Works flawlessly for me now too, no need to manually do `pods install` anymore :)

**Comment by Vikingjonsson** [comment]
I still get the error during Xcode build.

```
PhaseScriptExecution [CP]\ Check\ Pods\ Manifest.lock /Users/vikjon/Library/Developer/Xcode/DerivedData/Time_banding_party_planer-gyoxpiazcmmlezbvkylugqdxnocu/Build/Intermediates.noindex/Time\ banding\ party\ planer.build/Debug-iphonesimulator/Time\ banding\ party\ planer.build/Script-488F80062BA4C752AD98090B.sh (in target 'Time banding party planer' from project 'Time banding party planer')
    cd /Users/vikjon/Downloads/Packages/Time\ banding\ party\ planer.ios.project
    /bin/sh -c /Users/vikjon/Library/Developer/Xcode/DerivedData/Time_banding_party_planer-gyoxpiazcmmlezbvkylugqdxnocu/Build/Intermediates.noindex/Time\\\ banding\\\ party\\\ planer.build/Debug-iphonesimulator/Time\\\ banding\\\ party\\\ planer.build/Script-488F80062BA4C752AD98090B.sh

diff: /Podfile.lock: No such file or directory
error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
```

Tried runnig pod install and pods are updated to version 1.12.
Tested multiple projects.

**Comment by AshleyScirra** [comment]
There are three confirmations it's working now including my own, so I think the original issue is now resolved. @Vikingjonsson - I would guess that is a different issue which you'd need to report separately if you think it's a problem with Construct.

**Comment by Grizzley1996** [comment]
@AshleyScirra  I still have the same problem with installed pods and that isn't the different type of issue. Have tested with several project and still no result.

**Comment by AshleyScirra** [comment]
I'm afraid running `pod install` automatically on the build server ended up causing a serious problem (#6944). It seems it basically breaks projects using the Browser plugin, due to something weird about `cordova-plugin-inappbrowser`. I've not been able to figure out why or what workaround could be used, so I've reverted the change, and the build server no longer automatically runs `pod install`. I think for now plugins using cocoapods will need to document that the user needs to have cocoapods installed and run `pod install` themselves on the resulting project locally.

**Comment by Chadori** [comment]
That's too bad, although I understand.

**Comment by therealPaulPlay** [comment]
That is very unfortunate and weird. I‘ve been using the browser plugin in my project all along for opening links and vibrations and that has been working perfectly since the change which thankfully fixed the pods not being created for me.
What exactly is not working with the browser plugin? 
I hope there is an alternative solution to this…

**Comment by therealPaulPlay** [comment]
Also, the browser plugin does not create an empty pods folder for me.

**Comment by therealPaulPlay** [comment]
Maybe it works in my case, because there are other plugins that create a pod folder as well? Could you make it so that it only runs pod install if there is a plugin that needs it apart from the browser plugin?

**Comment by Chadori** [comment]
Hi @AshleyScirra, I've noticed that you use the `admob-plus-cordova@1.28.0` and the pods are installed when you use it, however it doesn't when we use it through the Addon SDK.

Upon investigating, it seems that the difference is `construct-mobile-export2@1.0.5` and `<preference name="DoCocoaPodsUpdate" value="true" />` which aren't exposed in the Addon SDK.

I've tested and it works if I manually add those values, except for an ordering bug which I am yet to solve, however I am getting close.

Would it be possible to expose these in the Addon SDK so that we can build iOS Cordova plugins that have Pod dependencies?
